File: CoreJava/Shoppingcart-app/src/com/techlabs/shoppingcart/Product.java
package com.techlabs.shoppingcart;

public class Product {
	private int id;
	private String name;
	private double cost;
	private float discount;

	public Product(int id, String name, double cost, float discount) {
		super();
		this.id = id > 0 ? id : 0;
		this.name = name;
		this.cost = cost > 0 ? cost : 0;
		this.discount = discount > 0 ? discount : 0f;
	}

	public int getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public double getCost() {
		return cost;
	}

	public float getDiscount() {
		return discount;
	}

	public double calcDiscountCost() {
		return cost > discount ? cost - (double) discount : cost;
	}

}

File: DotNET/C#/ShoppingCartApp/ShoppingCartApp/Product.cs
using System;

namespace ShoppingCartApp
{
    class Product
    {
        private int _id;
        private String _name;
        private double _cost;
        private float _discount;

        public Product(int id, String name, double cost, float discount)
        {
            _id = id;
            _name = name;
            _cost = cost;
            _discount = discount;
        }

        public int Id
        {
            get
            {
                return _id;
            }
        }

        public double Cost
        {
            get
            {
                return _cost;
            }
        }

        public string Name
        {
            get
            {
                return _name;
            }
        }

        public float Discount
        {
            get
            {
                return _discount;
            }
        }

        public double calcDiscountCost()
        {
            return _cost > _discount ? _cost - (double)_discount : _cost;
        }
    }
}
